src/java_bytecode/java_class_loader.h

```diff
@@ -28,6 +28,16 @@ class java_class_loadert:public messaget
   void set_java_cp_include_files(std::string &);
   void add_load_classes(const std::vector<irep_idt> &);
 
+  /// A function that yields a list of extra dependencies based on a class name.
+  typedef std::function<std::vector<irep_idt>(const irep_idt &)>
+    get_extra_class_refs_functiont;
+
+  /// Sets a function that provides extra dependencies for a particular class.
+  /// Currently used by the string preprocessor to note that even if we don't
+  /// have a definition of core string types, it will nontheless give them
+  /// certain known superclasses and interfaces, such as Serializable.
+  void set_extra_class_refs_function(get_extra_class_refs_functiont func);
```

Comment:
This is used in test-gen presumably (not used in this PR) - could you link the TG pointer bump PR

Reply:
It should have been, but I split this from its successor incorrectly. Now fixed.

Reply:
The pains of PR-stacks...